ping.py

```diff
@@ -63,19 +65,121 @@ def get_ident(): return 0
 MAX_SLEEP = 1000
 
 
-class MStats:
-    thisIP = "0.0.0.0"
-    pktsSent = 0
-    pktsRcvd = 0
-    minTime = 999999999
-    maxTime = 0
-    totTime = 0
-    avrgTime = 0
-    fracLoss = 1.0
+# noinspection PyAttributeOutsideInit
+class MStats2(object):
+
+    def __init__(self):
+        self._this_ip = '0.0.0.0'
+        self.reset()
+
+    def reset(self):
+        self._timing_list = []
+        self._packets_sent = 0
+        self._packets_rcvd = 0
+
+        self._reset_statistics()
+
+    @property
+    def thisIP(self):
+        return self._this_ip
+
+    @thisIP.setter
+    def thisIP(self, value):
+        self._this_ip = value
+
+    @property
+    def pktsSent(self):
+        return self._packets_sent
+
+    @property
+    def pktsRcvd(self):
+        return self._packets_rcvd
+
+    @property
+    def pktsLost(self):
+        return self._packets_sent - self._packets_rcvd
+
+    @property
+    def minTime(self):
+        return min(self._timing_list) if self._timing_list else None
+
+    @property
+    def maxTime(self):
+        return max(self._timing_list) if self._timing_list else None
+
+    @property
+    def totTime(self):
+        if self._total_time is None:
+            self._total_time = sum(self._timing_list)
+        return self._total_time
+
+    def _get_mean_time(self):
+        if self._mean_time is None:
+            if len(self._timing_list) > 0:
+                self._mean_time = self.totTime / len(self._timing_list)
+        return self._mean_time
+    mean_time = property(_get_mean_time)
+    avrgTime = property(_get_mean_time)
+
+    @property
+    def median_time(self):
+        if self._median_time is None:
+            self._median_time = self._calc_median_time()
+        return self._median_time
+
+    @property
+    def pstdev_time(self):
+        """Returns the 'Population Standard Deviation' of the set."""
+        if self._pstdev_time is None:
+            self._pstdev_time = self._calc_pstdev_time()
+        return self._pstdev_time
+
+    @property
+    def fracLoss(self):
+        if self._frac_loss is None:
+            if self.pktsSent > 0:
+                self._frac_loss = self.pktsLost / self.pktsSent
+        return self._frac_loss
+
+    def packet_sent(self, n=1):
```

Comment:
No, because those three are meant to be called() as methods?

It's `packet_sent` (1 or more packets have just been sent) not `packets_sent` (total number of packets sent so far); the latter is accessible through the `pktsSent` property.